### ISSUE TITLE
Thunderbird 68 Update - Features Discussion

### ISSUE THREAD

**Issue by cleidigh** [thundernest/import-export-tools#1]

**Comment by Sancus** [comment]
Here's how I would personally prioritize Import Export Tools functions:

1. Minimum Viable
    - Whole-Profile Backup
    - Whole-Profile Restore
    - [Backup Scheduling](https://i.imgur.com/zSq2l2y.png)
2. Nice-to-Have
    - Right-Click Copy Message/Headers to Clipboard(this is **really** useful, maybe Minimum Viable).
    - Folder Export (as mbox file only)
    - Folder Import (mbox)
    - Export messages (as csv, as "plaintext", EML)
        - With HTML index.
        - Just index.
3. All the other stuff
    - Filename options for message export.
    - [Search and Export window](https://i.imgur.com/CvRWtf2.png)
    - Import messages(EML only?).
    - Import/Export of formats not listed above:
        - Folders: [zipped, structured, etc](https://i.imgur.com/yw03Wej.png)?
        - Messages: PDF, HTML, with attachments, etc.

I'm not totally sure how to prioritize the various import and export file formats. For example, maybe it's really easy to export messages as text, but MUCH more complicated to import them back. That kind of thing. So the import/export options under #2 and #3 should be reprioritized based on apparent difficulty to whomever is implementing them.

We may want to try to ask a wider set of users how they use this add-on. For example, the import of individual messages is not useful to me, but there could be a lot of people using it for some strange workflow.

**Comment by jorgk3** [comment]
Sorry, your "minimum viable set" is a straight file system copy of the profile, maybe with a sync tool. I synchronise TB mail between three machines and never used "Import/Export Tools".

**Comment by Sancus** [comment]
> Sorry, your "minimum viable set" is a straight file system copy of the profile, maybe with a sync tool. I synchronise TB mail between three machines and never used "Import/Export Tools".

Sure, so? Interacting with the file system on a hidden directory(which is what the profile dir is) is not something we should ever be telling users to do. And telling them to use some third party sync tool is completely unacceptable.

**Comment by jorgk3** [comment]
After some discussion with Andrei on IRC, I formed this opinion: It might be useful to allow backing up and restoring "settings", like account definitions, preferences, etc. It might also be OK to allow the backup of local folders, maybe into a ZIP archive. Backing up IMAP data is questionable and restoring it can cause a real mess since the folder would get out of sync with the server. This basic functionality should be in "TB proper" and not some add-on.

**Comment by Sancus** [comment]
> Backing up IMAP data is questionable and restoring it can cause a real mess since the folder would get out of sync with the server.

Yes, I agree direct backup and restore of IMAP folders is dangerous and shouldn't be supported. I think that it's reasonable to export either an IMAP folder or a Local folder, but the Import side should only ever import as Local. If people really want to move mails back into their IMAP folders for some reason, they can copy them in TB.

**Comment by kewisch** [comment]
Is it really dangerous? I mean as long as you back up with all the state it should be ok? Use case would be for those with slow connections or bandwidth. Would you really like to redownload gigabytes of Email when migrating to a new computer?

**Comment by Sancus** [comment]
> Is it really dangerous? I mean as long as you back up with all the state it should be ok? Use case would be for those with slow connections or bandwidth. Would you really like to redownload gigabytes of Email when migrating to a new computer?

I actually went back and tried this a bunch since I hadn't used it recently [and then I looked at the code](https://github.com/thundernest/import-export-tools/blob/master/src/chrome/content/mboximport/profileImportWizard.js#L74). I don't see how what it's doing is dangerous at all. It's an automated file copy of the profile folder. On import, it copies it to a *new* folder and then prompts you to restart and select the "new copy" on your subsequent restart. It's only doing exactly what you would manually do on the file system to restore.

There are some steps that are probably a bit confusing for the average user(mainly the naming the new profile and selecting it on restart) that could probably be improved.

I think the weird/dangerous part would be trying to import a specific saved folder into an active IMAP account. But I will defer to the people actually familiar with TB code on this part :)

In any case, I do think file copy profile backup and restore is safe, and as you noted it is FAR faster than waiting to re-download your mail. Also saves a lot of bandwidth.

It is hard to see why we couldn't implement this in core in the future.
